step-release-vis/src/app/services/environment.ts

```diff
@@ -59,6 +59,30 @@ export class EnvironmentService {
     return new Polygon(points, candidate);
   }
 
+  private closePolygons(
+    polys: Polygon[],
+    lower: Map<string, Point[]>,
+    upper: Map<string, Point[]>,
+    set: TimestampLowerBoundSet
+  ): TimestampLowerBoundSet {
+    const newSet: TimestampLowerBoundSet = new TimestampLowerBoundSet();
+
+    for (let i = 0; i < set.snapshot.length; i++) {
+      if (
+        set.snapshot[i].position ===
+        (i === set.snapshot.length - 1 ? 100 : set.snapshot[i + 1].position)
```

Comment:
why not having the loop to l-1 and put this part after the loop?

Reply:
done :)